### PR TITLE
CHE-1180: add ability to configure private docker registries

### PR DESCRIPTION
**Pull request by olexii4**
@ashumilova WDYT?

in profilePreferences

{dockerCredentials: credentialsBase64}

credentials[registryUrl] = {
      name: userName,
      email: userEmail,
      password: userPassword
};

selection_015
selection_013
selection_012

### CODE REVIEW

**Comment by TylerJewell** [comment]
It should be "Docker Registries" -- missing capitalization.

**Comment by TylerJewell** [comment]
I am not understanding the user scenario.  
1. If a user has many registries, how does the system know which registry to push / pull from?
2. What is the trigger in our system that causes a "docker login <user> <pass>"?

**Comment by mmorhun** [comment]
Sorry, I accidentally closed it

**Comment by mmorhun** [comment]
Hi Tyler
1) So, for now we have three operations which may require authorization: pull, push and build.
- push is used for pushing workspace's snapshots only. Registry to which snapshots will be pushed is defined in properties.
- pull. When user create workspace from custom image he references it in the Docker recipe:
  `FROM myregistry:5000/myRepo`
  If this referenced repo is a private one, a user will then have to save credentials to user prefs in User Dashboard. It is also possible to save registry configuration in che.properties.

2) System do not perform login at all. It sends credentials on each query if it needed (i.e. login for one query).

**Comment by mmorhun** [comment]
@olexii4 please use `AuthConfig` dto format for credentials

**Comment by olexii4** [comment]
selection_007

File: dashboard/src/app/navbar/navbar.html [hunk 1]
```diff
@@ -50,6 +54,17 @@
...
               </md-button>
             </md-list-item>
+            <md-list-item flex class="navbar-subsection-item">
+              <md-button nav-bar-selected flex che-reload-href
+                         che-on-long-touch="navbarCtrl.openLinkInNewTab('{{navbarCtrl.menuItemUrl.administration}}')"
+                         ng-href="{{navbarCtrl.menuItemUrl.administration}}" layout-align="left">
+                <div class="navbar-item" layout="row" layout-align="start center">
+                  <md-icon md-font-icon="navbar-icon material-design icon-ic_settings_24px"></md-icon>
+                  <span>Administration</span>
+                  <span class="navbar-number" ng-show="navbarCtrl.getProjectsNumber()">&nbsp;({{navbarCtrl.getProjectsNumber()}})</span>
```
Comment (akurinnoy) [inline dashboard/src/app/navbar/navbar.html]: it seems copy-paste error

**Pull request merged**